sklearn/calibration.py

```diff
@@ -814,7 +814,7 @@ def _sigmoid_calibration(
     else:
         prior0 = float(np.sum(mask_negative_samples))
         prior1 = y.shape[0] - prior0
-    T = np.zeros_like(y, dtype=np.float64)
+    T = np.zeros_like(y, dtype=predictions.dtype)
     T[y > 0] = (prior1 + 1.0) / (prior1 + 2.0)
     T[y <= 0] = 1.0 / (prior0 + 2.0)
```

Comment:
So I'm wondering if we should not always do later

```python
return loss, grad.astype(np.float64)
```

Reply:
I think you are right that this seems to be an issue with scipy < 1.11.2. I guess if we always do it ourselves it wouldn't make a difference just that in the later versions it will also be done on the scipy side again redudantly?

Reply:
Maybe adding a version check would clarify why we are doing it though.

Reply:
For sure, it would be best to avoid the additional check within the gradient computation. We could have a comment mentioning that if LBFGS starts to support `np.float32` then we should avoid this conversion.